src/main/java/org/mitre/synthea/world/agents/Provider.java

```diff
@@ -235,13 +272,63 @@ public static void loadProviders(String state, String abbreviation, String filen
         if ("Yes".equals(row.remove("emergency"))) {
           parsed.servicesProvided.add(Provider.EMERGENCY);
         }
-
+        
         // add any remaining columns we didn't explicitly map to first-class fields
         // into the attributes table
         for (Map.Entry<String, String> e : row.entrySet()) {
           parsed.attributes.put(e.getKey(), e.getValue());
         }
         
+        String city = parsed.city;
+        String address = parsed.address;
+
+        //int population = (int) location1.getPopulation(city);
+        
+        //TODO - create a map of specialty:#
+        
+        boolean found = false;
+        //only enter this if THERE IS AN INFO FILE
+        //if there is no info file or if the provider is not found in the info file
+        // make them one general clinician that's always chosen
+        if (csvSpecialties == null || !found) {
+          parsed.specialtyInfo.put("GENERAL PRACTICE", 1);
+          parsed.clinicianMap.put("GENERAL PRACTICE", 
+              generateClinicianList(parsed, 1, "GENERAL PRACTICE"));
+        } else {
+          for (Map<String,String> rowSpecialty : csvSpecialties) {
+            if (rowSpecialty.get("address").equals(address) && rowSpecialty.get("city").equals(city)
+                    && rowSpecialty.get("state").equals(abbreviation)) {
+              for (Map.Entry<String, String> e : rowSpecialty.entrySet()) {
+                // if the field is not any of the provider info
+                if (!e.getKey().equals("address") && !e.getKey().equals("city") 
```

Comment:
Someone could want to add a new unrelated column and not realize that they have to add new checks here. Not a hard and fast rule, but it's often safer to look for things you do want instead of excluding specific things you don't want. If you extract the list of specialties to a new class, you could add a getSpecialtiesList() or something and then loop over that here to pick the columns you want.